### PR TITLE
Heuristic pitch finding fix py2->3 error and improve center of mass finding

### PR DESCRIPTION
**Pull request by MrMondrian**
Resolves: (#ID-of-the-issue)
- [x] I passed the docker hub test, and images can be built successfully.
- [x] I passed the GitHub CI test, so rodan functionalities and jobs work.

Improves the performance of pitch finding and addresses some subtle bugs.

Change 1: the correct note is now assigned to pitches. Consistently in the MEI files produced in the past year, clefs were assigned fraction position (ex: 3.5 instead of 3). This is because in py2, the / operator does floor division by default whereas is py3, it does float division. I fixed this by explicitly using // instead of /.

Change 2: the subimage used to find the center of mass is different depending on the neume component. If it is a neume with stacked punctums like a podatus, it now targets the bottom left neume instead of taking an entire vertical slice. For virgas, it now grabs the note head instead of the entire note. I believe that previously it was assumed that the tail of the virgas would not significantly impact the location of the center of mass. However from my experience, this is not the case. Finally, when finding the subimage, it extends to the right by only 0.8 * average punctum size. This is done to avoid accidentally grabbing part of the next neume component.

Here is a picture visualizing the subimage and COM finding process.
pitches_debug

Here are results from Salzinnes, Einsideln, and MS73. The first two used non-interactive classifier hence the mistakes.
Screenshot from 2023-06-27 10-31-29
Screenshot from 2023-06-27 10-31-38
Screenshot from 2023-06-27 10-31-43
In the second last image, I misclassified the intervals of the scadicuses in IC which is why the pitches are misplaced. Had I not done that I believe the results would have been perfect.

### CODE REVIEW

File: rodan-main/code/rodan/jobs/heuristic_pitch_finding/PitchFinding.py [hunk 1]
```diff
@@ -131,19 +122,57 @@ def _x_projection_vector(self, glyph):
 
         if g.ncols > self.discard_size and g.nrows > self.discard_size:
             if g.ncols < self.avg_punctum:
-                this_punctum_size = g.ncols
+                this_punctum_size_cols = g.ncols
             else:
-                this_punctum_size = self.avg_punctum
+                this_punctum_size_cols = self.avg_punctum * 0.8
```
Comment (timothydereuse) [inline rodan-main/code/rodan/jobs/heuristic_pitch_finding/PitchFinding.py:127]: Where does this 0.8 come from? If it's a heuristic, it should be defined elsewhere in the file and documented in comments (what it does, why it's at 0.8, acceptable range)

Comment (timothydereuse) [inline rodan-main/code/rodan/jobs/heuristic_pitch_finding/PitchFinding.py:127]: Where does this 0.8 come from? If it's a heuristic, it should be defined elsewhere in the file and documented in comments (what it does, why it's at 0.8, acceptable range) - I see 0.8 referenced later but it's in a different method

File: rodan-main/code/rodan/jobs/heuristic_pitch_finding/PitchFinding.py [hunk 2]
```diff
@@ -479,16 +491,14 @@ def __glyph_type(g):
 
                 # find octave
                 if my_strt_pos <= clef_line:
-                    octave = 3 + int((clef_line - my_strt_pos + noteShift) / len(SCALE))
+                    octave = 3 + int((clef_line - my_strt_pos + noteShift) // len(SCALE))
```
Comment (timothydereuse) [inline rodan-main/code/rodan/jobs/heuristic_pitch_finding/PitchFinding.py:494]: Replace all of the //'s with explicit floor functions - also add a comment describing what exactly this math is doing, so whoever sees this code next does not have to spend the time figuring out what's going on (like you did!)

**Review by timothydereuse — CHANGES_REQUESTED** [review]

File: rodan-main/code/rodan/jobs/heuristic_pitch_finding/PitchFinding.py (hunk 1 above)
Comment (MrMondrian) [reply rodan-main/code/rodan/jobs/heuristic_pitch_finding/PitchFinding.py]: fixed in c33d35f478714ce6f4c2c2e937172914001a7334.

File: rodan-main/code/rodan/jobs/heuristic_pitch_finding/PitchFinding.py (hunk 2 above)
Comment (MrMondrian) [reply rodan-main/code/rodan/jobs/heuristic_pitch_finding/PitchFinding.py]: fixed in 48df766fcd4db298d81115d6a6199c12d207dd72.

**Review by timothydereuse — APPROVED** [review]

**Pull request merged**